spec/controllers/settings_controller_spec.rb

```diff
@@ -31,17 +31,17 @@
   end
 
   describe 'on PUT /settings/dashboard' do
-    let(:constraints) do
-      Constraint::GROUPS.each_with_object([]) do |group, c|
-        c.push FactoryBot.create(:constraint, group: group)
+    let(:dashboard_items) do
+      DashboardItem::GROUPS.each_with_object([]) do |group, d|
+        d.push(DashboardItem.where(group: group)[0])
       end
     end
 
     let(:dash_settings) do
-      enum = Constraint::GROUPS.to_enum
+      enum = DashboardItem::GROUPS.to_enum
 
-      enum.with_index.each_with_object({}) do |(group, index), c|
-        c[ group ] = constraints[ index ].key
+      enum.with_index.each_with_object({}) do |(group, index), d|
```

Comment:
I'm pretty sure this linter is wrong, since `enum` isn't a Hash. Still the original code seems confusing and could probably be simplified:

```ruby
    let(:dash_settings) do
      DashboardItem::GROUPS
        .map
        .with_index { |value, index| [value, dashboard_items[index].key] }
        .to_h
    end
```